airbyte-integrations/connectors/destination-bigquery/src/test-integration/java/io/airbyte/integrations/destination/bigquery/BigQueryDestinationAcceptanceTest.java

```diff
@@ -187,71 +177,37 @@ private List<JsonNode> retrieveRecordsFromTable(final String tableName, final St
 
   @Override
   protected void setup(final TestDestinationEnv testEnv) throws Exception {
-    if (!Files.exists(CREDENTIALS_PATH)) {
-      throw new IllegalStateException(
-          "Must provide path to a big query credentials file. By default {module-root}/" + CREDENTIALS_PATH
-              + ". Override by setting setting path with the CREDENTIALS_PATH constant.");
-    }
-
-    final String fullConfigAsString = Files.readString(CREDENTIALS_PATH);
-    final JsonNode credentialsJson = Jsons.deserialize(fullConfigAsString).get(BigQueryConsts.BIGQUERY_BASIC_CONFIG);
-    final String projectId = credentialsJson.get(CONFIG_PROJECT_ID).asText();
-    final String datasetLocation = "US";
-
     final String datasetId = Strings.addRandomSuffix("airbyte_tests", "_", 8);
+    config = BigQueryDestinationTestUtils.createConfig(Path.of("secrets/credentials-standard.json"), datasetId);
+    bigquery = null;
+    dataset = null;
+    bqTornDown = false;
 
-    config = Jsons.jsonNode(ImmutableMap.builder()
-        .put(CONFIG_PROJECT_ID, projectId)
-        .put(CONFIG_CREDS, credentialsJson.toString())
-        .put(CONFIG_DATASET_ID, datasetId)
-        .put(CONFIG_DATASET_LOCATION, datasetLocation)
-        .build());
+    setUpBigQuery(config, datasetId);
+    addShutdownHook();
+  }
 
-    setupBigQuery(credentialsJson);
+  protected void setUpBigQuery(JsonNode config, String datasetId) throws IOException {
+    String projectId = config.get(BigQueryConsts.CONFIG_PROJECT_ID).asText();
+    bigquery = BigQueryDestinationTestUtils.initBigQuery(config, projectId);
+    try {
+      dataset = BigQueryDestinationTestUtils.initDataSet(config, bigquery, datasetId);
+    } catch(Exception ex) {
+      //ignore
+    }
   }
 
-  protected void setupBigQuery(final JsonNode credentialsJson) throws IOException {
-    final ServiceAccountCredentials credentials = ServiceAccountCredentials
-        .fromStream(new ByteArrayInputStream(credentialsJson.toString().getBytes(StandardCharsets.UTF_8)));
-
-    bigquery = BigQueryOptions.newBuilder()
-        .setProjectId(config.get(CONFIG_PROJECT_ID).asText())
-        .setCredentials(credentials)
-        .build()
-        .getService();
-
-    final DatasetInfo datasetInfo =
-        DatasetInfo.newBuilder(getDefaultSchema(config)).setLocation(config.get(CONFIG_DATASET_LOCATION).asText()).build();
-    dataset = bigquery.create(datasetInfo);
-
-    tornDown = false;
-    Runtime.getRuntime()
-        .addShutdownHook(
-            new Thread(
-                () -> {
-                  if (!tornDown) {
-                    tearDownBigQuery();
-                  }
-                }));
+  protected void addShutdownHook() {
+    Runtime.getRuntime().addShutdownHook(new Thread(() -> {
```

Comment:
Hm.. I thought it was already called after each test, because it overwrites `tearDown`. I'll add `@AfterEach` annotation. The shutdown hooks were there before refactoring and AFAIK, the reason is to execute cleanup even if the test crashes.

Reply:
I'm pretty sure junit takes care of running AfterEach methods, even if tests blow up (except maybe in cases where the entire runtime crashes?)

but yeah, don't worry about this if it looks super messy. Sounds like this has been working correctly as a shutdown hook anyway